### ISSUE TITLE
Cannot read property 'toString' of null

### ISSUE THREAD

**Issue by CharlieEtienne** [mikerourke/transfermyti.me#19]
Hi ! 

First of all, this project is a great idea!

Using the web interface at https://transfermyti.me, I encouter this error: 

The tool find my projects, and clients, but no time entries are found in any of those.

Steps: 

1. Select "Transfer from Toggl to Clockify"
2. Enter API Keys
3. Select Workspaces (I only have one)
4. Click next
5. Watch spinner saying it is fetching all sort of things
6. The following error occurred: `Cannot read property 'toString' of null`
7. All my projects and clients have 0 time entries

I looked into the console and I can find my time entries in json response of https://toggl.com/reports/api/v2/details page

I can send it to you in private message if it can help you testing

Charlie

**Comment by mikerourke** [comment]
@CharlieEtienne I suspect theres a property value coming back from the Toggl API as `null` that I'm not validating first. I'll push a fix that does null checks on all values that get converted to strings. Give me about an hour.

**Comment by CharlieEtienne** [comment]
@mikerourke Awesome!

Indeed, I may have in my toggl account:

- time entries that aren't in any project
- time entries that aren't related to any client or tag
- project without client
- client without project

I'm not sure what tasks are but I have none

Thanks!

**Comment by mikerourke** [comment]
@CharlieEtienne Fix is deployed! Could you reload the app and try it again to make sure the issue is fixed?

**Comment by CharlieEtienne** [comment]
@mikerourke I can see my time entries now! 
but when I start transfer, it finishes immediately with the success message on the page with the video and my clockify workspace is empty...

But I just realized I didn't verify my email address. I give it another try

**Comment by CharlieEtienne** [comment]
nope, it didn't transfer anything 😢

**Comment by mikerourke** [comment]
Is it hanging on the transfer page? I'm trying to do a transfer between my test accounts (from Toggl to Clockify) and so far it has only transferred 1 client in like 3 minutes.

**Comment by mikerourke** [comment]
Yikes, I just tried making a Clockify request through Postman and it is taking forrreeeverrrr! Maybe try again in an hour or so? I suspect there's something going on with Clockify's servers.

**Comment by CharlieEtienne** [comment]
I will try tomorrow because in France it's time to bed :)  

But I didn't see any progress bar or something. After clicking on start, it instantly displays this page https://transfermyti.me/tool-action-success, like nothing really happened...

**Comment by mikerourke** [comment]
Keep me posted. I accidentally made a field a string when it should have been a number on the fix I just merged, so creating projects in Clockify was failing. I just pushed a fix and deployed it. I'd suggest using the tool to delete any existing Clockify data (if any) was transferred and giving it another shot. If it still fails, we can take this out of GitHub to see what kind of data you're getting back. I apologize for the delay, have a good night!

**Comment by CharlieEtienne** [comment]
Hi, it still doesn't work. It has the same behaviour I described above. (no progression and instantly finished)

Another little thing: When you hit Ctrl+Shift+R (refresh and empty cache) on https://transfermyti.me/pick-tool-action page, it shows a Page not Found page. And the Back to our site link doesn't work.

**Comment by mikerourke** [comment]
@CharlieEtienne The back to our site issue is  a Netlify thing I still need to fix. For now, you have to manually navigate back to transfermyti.me. Could you try transferring just Projects or Tags? I’m curious if transferring something not contingent on a project will work.

**Comment by CharlieEtienne** [comment]
@mikerourke I unchecked all except projects and the same thing happened, meaning nothing...

**Comment by mikerourke** [comment]
Could you try just transferring projects again, but before hitting the Start button, open up DevTools and the Network tab? Once you hit Start, you should see a bunch of POST requests. If not, there’s something I screwed up. If you are seeing them, there’s probably something else I screwed up.

**Comment by CharlieEtienne** [comment]
![2020-01-22_17h08_16](https://user-images.githubusercontent.com/9772943/72911118-d4275880-3d39-11ea-906b-7ad3198823ad.png)

No POST requests, that's all I've got

I can give you my api keys in PM if you need them

**Comment by mikerourke** [comment]
That would probably be the quickest route to getting this issue fixed. Could you email them to me at mike.w.rourke@gmail.com?

**Comment by mikerourke** [comment]
@CharlieEtienne!!!! I think I fixed it! The issue was that your Toggl workspace and Clockify workspace names didn't match. I did a test transfer with just clients and it worked! The tool will actually create a Clockify workspace for you (if it doesn't exist). I should probably add a feature to allow a user to map a source workspace to a target workspace.

So right now you have 2 Clockify workspaces (one of them has nothing in it and the other one only has clients). Let me know if you're able to transfer the rest of your stuff.
